typescript/agentkit/src/action-providers/aave/aaveActionProvider.ts

```diff
@@ -0,0 +1,452 @@
+/* eslint-disable jsdoc/require-param-description */
+/* eslint-disable jsdoc/require-returns */
+/* eslint-disable jsdoc/require-description */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/member-ordering */
+import { Pool, EthereumTransactionTypeExtended } from "@aave/contract-helpers";
+import { AaveV3Base } from "@bgd-labs/aave-address-book";
+import { parseUnits, formatUnits } from "viem";
+import { z } from "zod";
+import { ActionProvider } from "../actionProvider";
+import { CreateAction } from "../actionDecorator";
+import { Network } from "../../network";
+import { EvmWalletProvider } from "../../wallet-providers";
+import { AAVE_V3_ADDRESSES, SUPPORTED_ASSETS } from "./constants";
+import { SupplySchema, WithdrawSchema } from "./schemas";
+import { approve } from "../../utils";
+import { ethers } from "ethers";
+
+export const SUPPORTED_NETWORKS = ["base-mainnet"];
+
+/**
+ * AaveActionProvider provides actions for interacting with the Aave protocol
+ */
+export class AaveActionProvider extends ActionProvider<EvmWalletProvider> {
+  /**
+   * Creates a new instance of AaveActionProvider
+   */
+  constructor() {
+    super("aave", []);
+  }
+
+  /**
+   * Check if network is supported
+   *
+   * @param network - The network to check support for
+   * @returns True if network is supported
+   */
+  supportsNetwork = (network: Network): boolean =>
+    network.protocolFamily === "evm" && SUPPORTED_NETWORKS.includes(network.networkId!);
+
+  /**
+   * Supply assets to Aave
+   *
+   * @param wallet - The wallet provider to use
+   * @param args - The supply parameters
+   * @returns A string describing the result
+   */
+  @CreateAction({
+    name: "supply",
+    description: `
+This tool will supply assets to the Aave protocol. It takes:
+- chain: The chain to supply on (e.g. base-mainnet)
+- amount: The amount to supply
+- asset: The asset to supply (e.g. ETH, USDC)
```

Comment:
It would be good to include the expected format of the asset if supplying ETH – will it be in ether, wei or gwei?